tests/kafkatest/tests/core/security_rolling_upgrade_test.py

```diff
@@ -100,17 +103,58 @@ def roll_in_sasl_mechanism(self, security_protocol, new_sasl_mechanism):
         # Bounce again with ACLs for new mechanism
         self.set_authorizer_and_bounce(security_protocol, security_protocol)
 
+    @cluster(num_nodes=9)
+    @matrix(broker_protocol=[SecurityConfig.SSL, SecurityConfig.SASL_PLAINTEXT, SecurityConfig.SASL_SSL],
```

Comment:
Do we need all these combinations of tests? Also, it is unrelated to upgrade, so could be in a different test class?

Reply:
Good point, I don't think we need all combinations here - we're interested in the following cases:
- client and broker on the different protocols
- client and broker on the same protocol, different sasl mechanisms
- client and broker on the same protocol, same sasl mechanism.
I'll update the PR with this change, please let me know if I'm missing something here.

Reply:
Changed the tests as you suggested, moving out interbroker listener tests into a separate test class and reducing their number.